ras_party/controllers/gov_uk_notify.py

```diff
@@ -1,45 +1,48 @@
-from notifications_python_client import NotificationsAPIClient
 from structlog import get_logger
 
 from ras_party.support.ras_error import RasNotifyError
+from ras_party.support.requests_wrapper import Requests
+from urllib import parse as urlparse
 
 log = get_logger()
 
 
 class GovUkNotify:
     """ Gov uk notify class"""
 
-    CLIENT_CLASS = NotificationsAPIClient
-
     def __init__(self, config):
         self.config = config
         self.notify_config = config.dependency['notify-service']
-        notify_keys = 'key-name-{}-{}'.format(self.notify_config['service_id'],
-                                              self.notify_config['api_key'])
-        self.notifications_client = self.CLIENT_CLASS(notify_keys)
 
     def _send_message(self, email, template_id, personalisation=None, reference=None):
         """
-        Send message to gov.uk notify
+        Send message to gov.uk notify wrapper
         :param email: email address of recipient
         :param template_id: the template id on gov.uk notify to use
         :param personalisation: placeholder values in the template
-        :param reference: reference to generated if not using Notify's id
+        :param reference: reference to be generated if not using Notify's id
         :rtype: 201 if success
         """
 
         if not self.config.feature.send_email_to_gov_notify:
```

Comment:
Do we even need this logic here? Does this not exist in `notifygatewaysvc`?

Reply:
Would you not want a toggle in this service even if there is one in notifygatewaysvc?

Reply:
I don't think so. The only scenario I can think of is in unit tests where you can just mock the request anyway?

Reply:
In the current logic it shouldn't be needed.
Currently we don't check the response from the notify service so if anything failed there, (500 error or for some other reason it fails to send to gov.notify) then we'd have no way of knowing, there's not even a log line for it I don't think.
If we had that check (which I think we should) this kind of toggle would help get around that when we run party locally

Reply:
Even then we can spin up a docker instance of a mock api which returns 200 which is defined in a docker-compose or something. It's always best to avoid putting hacks in production code to support tests or code to test locally. 

It leaves risk to running these things in production like in this scenario not sending emails when it should.

It removes unnecessary environmental/config variables used to support testing/local usage

Reply:
Yeah makes sense, I do think we should be checking the response from the notify service at some point in party though

Reply:
I'm not sure if this is now defunct. I didn't think it was in for testing but to stop in different cloud foundry environments. Though as you say I think the notifygatewaysvc only has the keys to hit gov notify in prod. I'll double check with @LJBabbage or @pricem14pc if this is the case.

Reply:
@tej99 might be one for the tech debt rather than here

Reply:
Agreed, if there is logic in the other service, we no longer need it here

Reply:
@benjefferies I'm leaving this toggle in as currently there isn't one available in the rm-notify gateway.

Reply:
Sure something we can add in as an improvement later